sync/optimistic.md

```diff
@@ -270,6 +270,23 @@ optimistic blocks (and vice-versa).
 
 ## Design Decision Rationale
 
+### Why sync optimistically?
+
+Most execution engines use state sync as a default sync mechanism on Ethereum Mainnet 
```

Comment:
Block sync also requires a cryptoeconomic head to do reverse header sync and then forward block sync.

I might make this more generic about "execution engine sync methods" needing cryptoeconomic heads to be able to sync independently from the consensus engine